packages/ui/src/common/components/Search/SearchResultsModal.tsx

```diff
@@ -19,16 +19,18 @@ import { ModalWithDataCall } from '@/common/providers/modal/types'
 import { ThreadItemBreadcrumbs } from '@/forum/components/threads/ThreadItemBreadcrumbs'
 import { ForumRoutes } from '@/forum/constant'
 
+import { useDebounce } from '../../hooks/useDebounce'
+
 export type SearchResultsModalCall = ModalWithDataCall<'SearchResults', { search: string }>
 
 export const SearchResultsModal = () => {
   const { hideModal, modalData } = useModal<SearchResultsModalCall>()
-
   const [search, setSearch] = useState(modalData.search)
   const [activeTab, setActiveTab] = useState<SearchKind>('FORUM')
   const { forum, forumPostCount, isLoading } = useSearch(search, activeTab)
   const pattern = useMemo(() => (search ? RegExp(escapeStringRegexp(search), 'ig') : null), [search])
-
+  const debouncedSearch = useDebounce(search, 400)
+  const isValid = () => !debouncedSearch || debouncedSearch.length === 0 || debouncedSearch.length > 2
```

Comment:
Great idea to debounce the search, but the pattern should use only valid `debouncedSearch` too. I think this should work:
```ts
  const debouncedSearch = useDebounce(search, 400)
  const isValid = () => !debouncedSearch || debouncedSearch.length === 0 || debouncedSearch.length > 2
  const pattern = useMemo(() => (isValid && search ? RegExp(escapeStringRegexp(search), 'ig') : null), [debouncedSearch])
```